cli/cmd/sync_v3.go

```diff
@@ -118,6 +123,9 @@ func syncConnectionV3(ctx context.Context, source v3source, destinations []v3des
 	// https://github.com/golang/go/issues/41087
 	syncTime := time.Now().UTC().Truncate(time.Microsecond)
 	sourceName := sourceSpec.Name
+	if shard != nil {
+		sourceName = fmt.Sprintf("%s-%d/%d", sourceName, shard.num, shard.total)
```

Comment:
This is needed to support `overwrite-delete-stale` otherwise shards delete each other's data at the end of the sync.

We can use a better name for it, maybe `%s-shard-%d/%d`